client/src/components/Notifications/Broadcasts/BroadcastContainer.vue

```diff
@@ -31,125 +37,266 @@ const props = defineProps<{
     options: Options;
 }>();
 
-const router = useRouter();
 const broadcastsStore = useBroadcastsStore();
-const { activeBroadcasts } = storeToRefs(broadcastsStore);
 const { renderMarkdown } = useMarkdown({ openLinksInNewPage: true });
 
-const remainingBroadcastsCountText = computed(() => {
-    const count = activeBroadcasts.value.length - 1;
-    return count > 0 ? `${count} more` : "";
+const multiple = computed(() => {
+    if (props.options.previewMode) {
+        return false;
+    } else {
+        return props.options.broadcasts.length > 1;
+    }
+});
+
+const page = ref(0);
+
+const currentPage = computed({
+    get: () => {
+        return page.value;
+    },
+    set: (newPage) => {
+        page.value = newPage;
+        checkPageInBounds();
+    },
 });
 
-function getBroadcastVariant(item: { variant: string }) {
-    switch (item.variant) {
-        case "urgent":
-            return "danger";
-        default:
-            return item.variant;
+function checkPageInBounds() {
+    if (page.value < 0) {
+        page.value = sortedBroadcasts.value.length - 1;
+    } else if (page.value >= sortedBroadcasts.value.length) {
+        page.value = 0;
     }
 }
 
-function onActionClick(item: BroadcastNotification, link: string) {
-    if (link.startsWith("/")) {
-        router.push(link);
+const displayedBroadcast = computed(
+    () => ensureDefined(sortedBroadcasts.value[currentPage.value]) as BroadcastNotification
+);
+
+function sortByImportanceAndPublicationTime(a: BroadcastNotification, b: BroadcastNotification) {
+    if (a.variant === "urgent" && b.variant !== "urgent") {
```

Comment:
This condition seems contradicting or I'm missing something :thinking:

Reply:
> If a is urgent, and b is not urgent, put a before b.

It's equivalent to:

> If a is urgent, and b is a warning or an info, put a before b.

I don't think there's any contradiction here.

Reply:
Lol! I was missing something then... I didn't notice the `b` and was reading `a` in both cases... sorry for the noise I need more coffee :sweat_smile: